manifests/init.pp

```diff
@@ -5,33 +5,39 @@
 # @param take_ownership Enable to allow Puppet to take ownership
 #   of the TPM.
 #
+# @param tpm_name  The name of the device (usually tpm0).
+#
+# @param tpm_version  Override for the tpm_version fact.
+#
+# @param ensure  The ensure status of packages to be installed
+#
 # @author Nick Markowski <nmarkowski@keywcorp.com>
 # @author Nick Miller <nick.miller@onyxpoint.com>
 #
 class tpm (
-  Boolean $ima            = false,
-  Boolean $take_ownership = false
+  Boolean          $ima            = false,
+  Boolean          $take_ownership = false,
+  String           $tpm_name       = 'tpm0',
+  Optional[Enum[
+    'tpm1',
+    'tpm2',
+    'unknown']]    $tpm_version    = $facts['tpm_version'],
+  String           $ensure         = simplib::lookup('simp_options::package_ensure', { 'default_value' => 'installed' })
```

Comment:
Calling this `package_ensure` would make sure people knew what they were setting, to avoid confusion with other `ensure` params in this module